internal/commands/buildpack_new.go

```diff
@@ -0,0 +1,89 @@
+package commands
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/buildpacks/pack/internal/build"
+
+	"github.com/spf13/cobra"
+
+	"github.com/buildpacks/pack"
+	"github.com/buildpacks/pack/internal/dist"
+	"github.com/buildpacks/pack/internal/style"
+	"github.com/buildpacks/pack/logging"
+)
+
+// BuildpackNewFlags define flags provided to the BuildpackCreate command
+type BuildpackNewFlags struct {
+	API     string
+	Path    string
+	Stacks  []string
+	Version string
+}
+
+// BuildpackCreator creates buildpacks
+type BuildpackCreator interface {
+	NewBuildpack(ctx context.Context, options pack.NewBuildpackOptions) error
+}
+
+// BuildpackNew generates the scaffolding of a buildpack
+func BuildpackNew(logger logging.Logger, client BuildpackCreator) *cobra.Command {
+	var flags BuildpackNewFlags
+	cmd := &cobra.Command{
+		Use:     "new <id>",
+		Short:   "Creates basic scaffolding of a buildpack.",
+		Args:    cobra.ExactValidArgs(1),
+		Example: "pack buildpack new sample/my-buildpack",
+		Long:    "buildpack new generates the basic scaffolding of a buildpack repository. It creates a new directory `name` in the current directory (or at `path`, if passed as a flag), and initializes a buildpack.toml, and two executable bash scripts, `bin/detect` and `bin/build`. ",
+		RunE: logError(logger, func(cmd *cobra.Command, args []string) error {
+			id := args[0]
+			idParts := strings.Split(id, "/")
+			dirName := idParts[len(idParts)-1]
+			_, err := os.Stat("temp")
+			if !os.IsNotExist(err) {
+				return err
+			}
+
+			path := filepath.Join(flags.Path, dirName)
```

Comment:
I initially thought this was a bug (because if the `--path` flag wasn't provided, it would make it `/$dirName`, but apparently filepath.Join ignored empty strings entirely https://golang.org/pkg/path/filepath/#Join